machine/v1beta1/types_gcpprovider.go

```diff
@@ -143,10 +147,14 @@ type GCPMachineProviderSpec struct {
 	// +optional
 	ShieldedInstanceConfig GCPShieldedInstanceConfig `json:"shieldedInstanceConfig,omitempty"`
 
-	// confidentialCompute Defines whether the instance should have confidential compute enabled.
-	// If enabled OnHostMaintenance is required to be set to "Terminate".
+	// ConfidentialCompute Defines whether the instance should have confidential compute enabled or not, and the confidential computing technology of choice.
+	// If Disabled, the machine will not be configured to be a confidential computing instance.
+	// If Enabled, confidential computing will be configured and AMD Secure Encrypted Virtualization will be configured by default. That is subject to change over time. If using AMD Secure Encrypted Virtualization is vital, use AMDEncryptedVirtualization explicitly instead.
```

Comment:
Maybe something like:
> When set to Enabled, the machine will be configured as a confidential computing instance with no preference on the confidential compute policy used. In this mode, the platform chooses a default that is subject to change over time. Currently, the default is to use AMD Secure Encrypted Virtualization.

?

Reply:
I agree with the suggestion. I'm applying it. The current state is that it just does the same as `AMDEncryptedVirtualization` due to backward compatibility.

FYI: Instead of extending it to a value that would choose a default based on the confidential computing technology supported by the chosen machine series, it will probably be marked as deprecated at some point: https://github.com/kubernetes-sigs/cluster-api-provider-gcp/issues/1287#issuecomment-2678635686